pkg/kinflate/commands/secret.go

```diff
@@ -63,6 +78,35 @@ func newCmdAddSecretGeneric(errOut io.Writer) *cobra.Command {
 	return cmd
 }
 
+func addGenericSecret(m *manifest.Manifest, config dataConfig) error {
+	gs := getOrCreateGenericSecret(m, config.Name)
+
+	err := mergeData(&gs.DataSources, config)
+	if err != nil {
+		return err
+	}
+
+	// Validate manifest's generic secret by creating a generic secret.
+	_, _, err = configmapandsecret.MakeGenericSecretAndGenerateName(*gs)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getOrCreateGenericSecret(m *manifest.Manifest, name string) *manifest.GenericSecret {
+	for i, s := range m.GenericSecrets {
+		if name == s.Name {
+			return &m.GenericSecrets[i]
+		}
+	}
+	// config map not found, create new one and add it to the manifest.
+	gs := &manifest.GenericSecret{Name: name}
+	m.GenericSecrets = append(m.GenericSecrets, *gs)
```

Comment:
why do you get a pointer, if the goal is to push a copy? (that probably already applied to configmap ...)

Reply:
Fixed